Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -21,7 +21,10 @@ open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType
 
         // Creates an observable that starts a request each time it's subscribed to.
         return Observable.create { [weak self] observer in
```

Comment:
Here is another potential solution: We could support both behaviors.

In the instance method `request` we can weakly retain the provider assuming the user is handling that. We will emit the error if it was deallocated.

Then we can create a class method on `RxMoyaProvider` that will retain the provider in the request. Then we can still support workflows where users want ad-hoc requests out of the box. Something like:

```swift
RxMoyaProvider<MyApi>.request(/* ... */)
```